src/Database/Schema/MysqlSchemaDialect.php

```diff
@@ -115,6 +116,19 @@ protected function _convertColumn(string $column): array
             $precision = (int)$precision;
         }
 
+        if ($length === null && $comment && str_starts_with($comment, '[schema]')) {
```

Comment:
Is MySQL generating the comments on these columns when a database is upgraded from 5 to 8?

Reply:
No, I am currently writing a Command for it using DESCRIBE and the option of either applying any fix directly or generating a Cake migrations file for those tables.

Reply:
For Cake4 I hacked it to just directly modify: https://github.com/dereuromark/cakephp-setup/commit/0a37e591f5279984c2d9cdd8f32705c6368904fe#diff-049a4c2ac78c0d7254b3166847992b4eb4691c47743470460d0eaabf3cfceafcR65-R67
But for Cake5 I am looking more into actual migration file generation as preferred option.

Reply:
We also would want migrations (snapshot) to include those or the core to auto add those meta comments when running migrations without it set inside comment.
This way also re-running migrations on a later state (8+) will still work, even if you didnt manually upgrade beforehand.

Reply:
Do other databases like Postgres also have length for integer fields? If not then I don't think we should add such features based on the column comment. Instead those upgrading their mysql server should modify their schema as required.

Reply:
> Instead those upgrading their mysql server should modify their schema as required.

What does that mean?
One needs to have the ability to have the length present inside the application, for reasons outlined above.
So if the schema is not able to contain it, the length (even if arbitrary set) needs to be able to get injected via meta data.

Reply:
> Do other databases like Postgres also have length for integer fields?

Postgres only supports [three sizes of integers](https://www.postgresql.org/docs/current/datatype-numeric.html) (small, integer, bigint).

> Instead those upgrading their mysql server should modify their schema as required.

Updating either your schema as you upgrade or remapping types in the ORM seems like the more forward looking solution.

Reply:
> One needs to have the ability to have the length present inside the application..

Not really. Other databases like Postgres don't have a length for integer columns and people able to use it just fine. There must be many who use MySQL too without specifying explicit length for the integer columns and not have any problems.

Adding pseudo support for a non-standard defunct feature of a db is not a good long term strategy and would just add unnecessary maintenance burden.

Reply:
Could there be a way to allow meta data to be injected for all db types as plugin/hook solution?
Maybe by providing an extension point here?